lib/auth0/exception.rb

```diff
@@ -25,12 +41,18 @@ class MissingClientId < Auth0::Exception; end
   # exception for an unset parameter
   class MissingParameter < Auth0::Exception; end
   # Api v2 access denied
-  class AccessDenied < Auth0::Exception; end
+  class AccessDenied < Auth0::HTTPError; end
   # Invalid parameter passed, e.g. empty where ID is required
   class InvalidParameter < Auth0::Exception; end
   # Invalid Auth0 credentials either client_id/secret for API v1
   # or JWT for API v2/
   class InvalidCredentials < Auth0::Exception; end
   # Invalid Auth0 API namespace
   class InvalidApiNamespace < Auth0::Exception; end
+  # Auth0 API rate-limiting encountered
+  class RateLimitEncountered < Auth0::HTTPError
```

Comment:
Should we extend `Auth0::Unsupported` here instead of `Auth0::HTTPError`? Without that, we're not backwards-compatible (we're now throwing a different exception and, for apps that are rescuing `Auth0::Unsupported` in the case of a rate limit hit, that exception will bubble up).

Reply:
Ah, hm, interesting point. From an API-compatibility point, yes; from a semantic standpoint, no?

Actually, I wonder if `Auth0::Unsupported` should just become an alias for `Auth0::HTTPError`. After all, what _is_ `Auth0::Unsupported` if not "An HTTP error that this gem doesn't have any more specific classification for"?

E.g., instead of making `Auth0::RateLimitEncountered` inherit from `Auth0::Unsupported`, just have this line in there:

`Unsupported = Auth0::HTTPError`

Reply:
Hm. Actually, maybe that's not great either, since it would broaden the scope of `Auth0::Unsupported`; `rescue Auth0::Unsupported` would not rescue error types that it didn't previously.

Reply:
I'm more worried about breaking folks on a minor than semantics, at this point. We're due for a major on this SDK (a number of deprecated methods and addressing the HTTP client we're using) and could remove that extension then.

Reply:
Ok, I've made `Auth0::RateLimitEncountered` a subclass of `Auth0::Unsupported`, with a comment explaining the reason and a TODO item to move it at a major version change.